scripts/juju_inventory_script.py

```diff
@@ -30,18 +27,18 @@ def main():
         # This allows this script to be used as a dynamic inventory plugin.
         if not sys.argv[1:] or sys.argv[1] == '--list':
             # Build an Ansible inventory file from Juju environment status
-            username = os.getenv('JUJU_USERNAME')
-            password = os.getenv('JUJU_PASSWORD')
-            model_uuid = os.getenv('JUJU_MODEL_UUID')
-
-            url = '{0}?model_uuid={1}'.format(inventory_url, model_uuid)
-            req = urllib.request.Request(url)
-            credentials = '{0}:{1}'.format(username, password)
-            encoded_credentials = base64.b64encode(credentials.encode('ascii'))
-            req.add_header(
-                'Authorization',
-                'Basic {}'.format(encoded_credentials.decode('ascii')))
-            res = urllib.request.urlopen(req).read()
+            params = {
+                'juju': {
+                    'username': os.getenv('JUJU_USERNAME'),
+                    'password': os.getenv('JUJU_PASSWORD'),
+                    'model_uuid': os.getenv('JUJU_MODEL_UUID'),
+                    'cacert': os.getenv('JUJU_CACERT').replace(r'\n', '\n'),
+                    'endpoint': os.getenv('JUJU_ENDPOINT')
+                }
+            }
+            req = Request('{}/juju/inventory'.format(os.getenv('AIS_URL')),
```

Comment:
Introduction of AIS_URL should be documented IMO because we are expecting user to provide a value

Reply:
Updated docs regarding the API changes. Documentation is in dire need of a re-write, but this should be done on a separate pull request